src/com/trollworks/gcs/skill/SkillOutline.java

```diff
@@ -157,6 +158,71 @@ public void increment() {
 	}
 
 	@Override
+	public boolean canIncrementSkillLevel() {
+		return canIncrement();
+	}
+
+	@Override
+	public boolean canDecrementSkillLevel() {
+		return canDecrement();
+	}
+
+	@SuppressWarnings("unused")
+	@Override
+	public void incrementSkillLevel() {
+		List<RowUndo> undos = new ArrayList<>();
+		for (Skill skill : new FilteredIterator<>(getModel().getSelectionAsList(), Skill.class)) {
+			if (!skill.canHaveChildren()) {
+				int basePoints = skill.getPoints() + 1;
+				int maxPoints = basePoints + (skill.getDifficulty() == SkillDifficulty.W ? 12 : 4);
```

Comment:
What's the purpose of having a maximum points you will add? Given that you just loop until the skill level changes, this seems unnecessary. What am I missing?